include/oneapi/dpl/pstl/parallel_backend_omp.h

```diff
@@ -0,0 +1,221 @@
+// -*- C++ -*-
+// -*-===----------------------------------------------------------------------===//
+//
+// Copyright (C) Christopher Nelson
+//
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+//
+//===----------------------------------------------------------------------===//
+
+#ifndef _ONEDPL_PARALLEL_BACKEND_OMP_H
+#define _ONEDPL_PARALLEL_BACKEND_OMP_H
+
+#include <algorithm>
+#include <atomic>
+#include <iterator>
+#include <cstddef>
+#include <cstdio>
+#include <memory>
+#include <vector>
+#include <omp.h>
+
+#include "parallel_backend_utils.h"
+#include "unseq_backend_simd.h"
+#include "utils.h"
+
+#if !defined(_OPENMP)
+#    error _OPENMP not defined.
+#endif
+
+// Portability "#pragma" definition
+#ifdef _MSC_VER
+#    define _PSTL_PRAGMA(x) __pragma(x)
+#else
+#    define _PSTL_PRAGMA(x) _Pragma(#    x)
+#endif
+
+namespace oneapi
+{
+namespace dpl
+{
+namespace __omp_backend
+{
+
+//------------------------------------------------------------------------
+// use to cancel execution
+//------------------------------------------------------------------------
+inline void
+__cancel_execution()
+{
+}
+
+//------------------------------------------------------------------------
+// raw buffer
+//------------------------------------------------------------------------
+
+template <typename _ExecutionPolicy, typename _Tp>
+class __buffer
```

Comment:
I believe  we can share that class with TBB backend. (a new buffer class may be parametrized by allocator type - tbb::allocator or std::allocator).
In that case the code should be moved from omp headers, to into parallel_backend_utils.h, for example.

Reply:
Done.